src/Compilers/Core/CodeAnalysisTest/Analyzers/AnalyzerFileReferenceTests.cs

```diff
@@ -267,6 +267,49 @@ public void TestLoadGenerators()
             Assert.DoesNotContain("Microsoft.CodeAnalysis.UnitTests.NotAGenerator", typeNames);
         }
 
+        // can't load a coreclr targeting generator on net framework / mono
+        [ConditionalFact(typeof(CoreClrOnly))]
+        public void TestGeneratorsCantTargetNetFramework()
+        {
+            var directory = Temp.CreateDirectory();
+
+            // core
+            var errors = loadGeneratorAndReturnAnyErrors(TestResources.GeneratorTests.NetCoreGenerator);
+            Assert.Empty(errors);
+
+            // netstandard
+            errors = loadGeneratorAndReturnAnyErrors(TestResources.GeneratorTests.NetStandardGenerator);
+            Assert.Empty(errors);
+
+            // no target
+            errors = loadGeneratorAndReturnAnyErrors(TestResources.GeneratorTests.NoTargetGenerator);
+            Assert.Empty(errors);
+
+            // framework
+            errors = loadGeneratorAndReturnAnyErrors(TestResources.GeneratorTests.NetFrameworkGenerator);
+            Assert.Equal(1, errors.Count);
+            Assert.Equal(AnalyzerLoadFailureEventArgs.FailureErrorCode.ReferencesFramework, errors.First().ErrorCode);
+
+            List<AnalyzerLoadFailureEventArgs> loadGeneratorAndReturnAnyErrors(byte[] assembly)
+            {
+                var directory = Temp.CreateDirectory();
+
+                var generator = directory.CreateFile("generator.dll").WriteAllBytes(assembly);
+                AnalyzerFileReference reference = CreateAnalyzerFileReference(generator.Path);
+
+                List<AnalyzerLoadFailureEventArgs> errors = new List<AnalyzerLoadFailureEventArgs>();
+                void errorHandler(object? o, AnalyzerLoadFailureEventArgs e) => errors.Add(e);
+                reference.AnalyzerLoadFailed += errorHandler;
+
+                var builder = ImmutableArray.CreateBuilder<ISourceGenerator>();
+                reference.AddGenerators(builder, LanguageNames.CSharp);
+                var analyzers = builder.ToImmutable();
```

Comment:
>var analyzers = builder.ToImmutable(); [](start = 16, length = 38)

The statement appears to be unused. Should we `Assert.Equal(1, analyzers.Length)`?

Reply:
Honestly, that was just left over from copy paste, but it seems sensible to assert that we did actually load a generator, so will add.